### PR TITLE
 ignore check  if the queue item exists

### PR DESCRIPTION
**Pull request by ide-rea**
**What type of PR is this?**
 /kind cleanup
**What this PR does / why we need it**:
the queue  and  items are strictly 1:1 and the pop operation is protected by lock, so we don't need to check if item exists.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by lavalamp** [comment]
https://github.com/kubernetes/kubernetes/pull/91968 is already in my queue about this...

File: staging/src/k8s.io/client-go/tools/cache/fifo.go [hunk 1]
```diff
@@ -294,11 +294,8 @@ func (f *FIFO) Pop(process PopProcessFunc) (interface{}, error) {
 		if f.initialPopulationCount > 0 {
 			f.initialPopulationCount--
 		}
-		item, ok := f.items[id]
-		if !ok {
-			// Item may have been deleted subsequently.
-			continue
-		}
+
+		item := f.items[id]
```
Comment (mikedanese) [inline staging/src/k8s.io/client-go/tools/cache/fifo.go:298]: As mentioned, the comment is inccorect. See:

https://github.com/kubernetes/kubernetes/blob/b1b93e30131986319f1d46a62771feb9c0f097ae/staging/src/k8s.io/client-go/tools/cache/fifo.go#L210-L220

**Pull request closed without merging**